src/main/java/com/kodilla/ecommercee/domain/Product.java

```diff
@@ -20,6 +22,7 @@
 
 import static org.hibernate.annotations.FetchMode.SELECT;
 
+
```

Comment:
30. Hej. Wydaje mi się, że jak używasz adnotacji @Data, to nie musisz już używać @Getter i @Setter - można wyrzucić, będzie przejrzyściej.

46. W/g mnie pole String groupId jest niepotrzebne, masz pole Group group elegancko opisane adnotacją i hibernate sam utworzy pole z kluczem obcym "GROUP_ID"

48-61. Może wystarczyłaby jedna kolekcja obiektów CartItem? Albo Set, albo List, wydaje mi się że dwie są niepotrzebne.

73. Jak już nie będzie pola String groupId z linii 46, to ten konstruktor z linii 73 też już chyba nie będzie potrzebny, bo załatwi go adnotacja @AllArgsConstructor, którą umieściłeś u góry.

Jakbyś się zdecydował wprowadzać jakieś zmiany, to jakbyś mógł przy okazji zrobić jednolite formatowanie, tj. albo zostawiać zawsze wolną linię po każdym polu klasy, albo nie zostawiać po wszystkich, lepiej by się to czytało.

Poza tym jest OK, koncepcja całkiem w porządku.

Reply:
ok

Reply:
Konstruktor z linii 73 może się przydać, używam go do mapowania(pomija on set cartItem). Jednak dla przejrzystości można się go pozbyć. Co do reszty to oczywiście masz całkowitą rację. Ale to nie dotyczy tego pull requesta. Zmiany już są w projekcie i trzeba będzie to naprawić. Bałagan powstał przy Jdp221201 12.